<launch>
  <arg name="enable_raw" default="false"/>
  <arg name="enable_imv" default="true"/>
  <arg name="camera_id" default="0"/>

  <node type="raspicam_node" pkg="raspicam_node" name="raspicam_node" output="screen">
    <param name="private_topics" value="true"/>

    <param name="camera_frame_id" value="camera"/>
    <param name="enable_raw" value="false"/>
    <param name="enable_imv" value="true"/>
    <param name="camera_id" value="0"/>
    
    <!-- Убираем переназначение топика -->
    <!-- <remap from="/raspicam_node/image" to="/raspicam_node/image_raw" /> -->

    <param name="camera_name" value="main_camera_optical"/>
    <param name="camera_info_url" value="package://aruco_pose/fisheye_cam.yaml"/>
    <param name="width" value="640"/>
    <param name="height" value="480"/>

    <param name="framerate" value="30"/>
    <param name="exposure_mode" value="auto"/> <!-- Меняем на auto вместо antishake -->
    <param name="shutter_speed" value="0"/>
    
    <!-- Добавляем параметры для улучшения видимости -->
    <param name="brightness" value="55"/>
    <param name="contrast" value="50"/>
    <param name="saturation" value="50"/>
    <param name="ISO" value="800"/>
    <param name="video_stabilisation" value="true"/>
    
    <!-- Включаем сжатие изображения -->
    <param name="enable_compressed" value="true"/>
    <param name="compressed_image_transport" value="true"/>
  </node>
</launch>
